patterns/Prototype/Prototype.js

```diff
@@ -0,0 +1,20 @@
+// Prototype object
+const vehiclePrototype = {
+    init(make, model) {
+      this.make = make;
+      this.model = model;
+    },
+    getDetails() {
+      return `${this.make} ${this.model}`;
+    },
+  };
+  
+  // Create new instances using the prototype
+  const car1 = Object.create(vehiclePrototype);
+  car1.init('Toyota', 'Camry');
+  
+  const car2 = Object.create(vehiclePrototype);
+  car2.init('Honda', 'Civic');
+  
+  console.log(car1.getDetails()); // Output: 'Toyota Camry'
+  console.log(car2.getDetails()); // Output: 'Honda Civic'
```

Comment:
You could probably do a bit more to demonstrate that this conforms to the prototype pattern. In particular, you could compare the two car objects to demonstrate that they are distinct, but then also compare their respective `prototype` values (see more info below) to show that they are derived from the same base object:
```javascript
console.log(car1 === car2) //prints true
console.log(car1.prototype === car2.prototype) //prints false
```